errbot/backends/irc.py

```diff
@@ -248,6 +277,9 @@ def topic(self, topic):
         :param topic:
             The topic to set.
         """
+        if not self.joined:
+            raise RoomNotJoinedError("Must join the room to set the topic")
+        self.cb_set_topic(topic)
```

Comment:
No worries at all. Happy to review and help iterate on this until it's ready for merging! The IRC backend really needs some of this love so I'm very happy with these improvements you've been making to it.